### PR TITLE
MAILBOX-374 Enforce bounded parallelism for the EventBus 

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java [hunk 1]
```diff
@@ -56,6 +59,30 @@
...
+                    rateExceeded.set(true);
+                }
+                nbCalls.incrementAndGet();
+                Thread.sleep(Duration.ofMillis(200).toMillis());
+                finishedExecutions.incrementAndGet();
+
+            }, KEY_1);
+
+            IntStream.range(0, eventCount)
+                .forEach(i -> eventBus().dispatch(EVENT, KEY_1).block());
```
Comment (blackheaven) [inline mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: Wouldn't it be faster/more interesting to use a `Flux` (and do it in parallel)?

Comment (mbaechler) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: maybe but the simpler the better, right ? It's just a test

**Comment by mbaechler** [comment]
> You are enforcing it on rabbitmq event system. But you don't need to do the same on Cassandra DAOs?

It's done at the EventBus level so it has effects on every listeners.

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java (hunk 1 above)
Comment (blackheaven) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: ok

**Comment by rouazana** [comment]
merged

**Pull request merged**